src/pseudopeople/dataset.py

```diff
@@ -0,0 +1,154 @@
+                        and noise_type.name in noise_configuration.column_noise[col]
+                    ]
+                    # Apply column noise to each column as appropriate
+                    for column in columns_to_noise:
+                        noise_type(
+                            self,
+                            noise_configuration.column_noise[column][noise_type.name],
+                            column,
+                        )
+
+            else:
+                raise TypeError(
+                    f"Invalid noise type. Allowed types are {RowNoiseType} and "
+                    f"{ColumnNoiseType}. Provided {type(noise_type)}."
+                )
+
+    def _reformat_dates_for_noising(self) -> None:
+        """Formats date columns so they can be noised as strings."""
+        data = self.data.copy()
+
+        for date_column in [COLUMNS.dob.name, COLUMNS.ssa_event_date.name]:
+            # Format both the actual column, and the shadow version that will be used
+            # to copy from a household member
+            for column in [date_column, COPY_HOUSEHOLD_MEMBER_COLS.get(date_column)]:
+                if column in data.columns:
+                    # Avoid running strftime on large data, since that will
+                    # re-parse the format string for each row
+                    # https://github.com/pandas-dev/pandas/issues/44764
+                    # Year is already guaranteed to be 4-digit: https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#timeseries-timestamp-limits
+                    is_na = data[column].isna()
+                    data_column = data.loc[~is_na, column]
+                    year_string = data_column.dt.year.astype(str)
+                    month_string = _zfill_fast(data_column.dt.month.astype(str), 2)
+                    day_string = _zfill_fast(data_column.dt.day.astype(str), 2)
+                    if self.dataset.date_format == DATEFORMATS.YYYYMMDD:
+                        result = year_string + month_string + day_string
+                    elif self.dataset.date_format == DATEFORMATS.MM_DD_YYYY:
+                        result = month_string + "/" + day_string + "/" + year_string
+                    elif self.dataset.date_format == DATEFORMATS.MMDDYYYY:
+                        result = month_string + day_string + year_string
+                    else:
+                        raise ValueError(f"Invalid date format in {self.dataset.name}.")
+
+                    data[column] = pd.Series(np.nan, dtype=str)
+                    data.loc[~is_na, column] = result
+
+        self.data = data
+
+    def drop_extra_columns(self) -> None:
+        """Drops columns that are not in the dataset schema."""
+        self.data = self.data[[c.name for c in self.dataset.columns]]
+
+    @staticmethod
+    def is_missing(data: pd.DataFrame) -> pd.DataFrame:
+        """Returns the missingness of the data."""
+        return (data == "") | (data.isna())
```

Comment:
Is there a risk of empty spaces that need to be interpreted as missing? e.g. "   "